### PR TITLE
8307067: remove broken EnableThreadSMRExtraValidityChecks option

### PR DESCRIPTION
**Pull request by dcubed-ojdk**
A trivial fix to remove broken EnableThreadSMRExtraValidityChecks option.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8307067](https://bugs.openjdk.org/browse/JDK-8307067): remove broken EnableThreadSMRExtraValidityChecks option

### Reviewers
 * [Coleen Phillimore](https://openjdk.org/census#coleenp) (@coleenp - **Reviewer**) ⚠️ Review applies to [bfb24496](https://git.openjdk.org/jdk/pull/13704/files/bfb244966c3cdf7122ce4f361c50f3c3d4d05e2b)
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)
 * [Serguei Spitsyn](https://openjdk.org/census#sspitsyn) (@sspitsyn - **Reviewer**) ⚠️ Review applies to [bfb24496](https://git.openjdk.org/jdk/pull/13704/files/bfb244966c3cdf7122ce4f361c50f3c3d4d05e2b)
 * [Robbin Ehn](https://openjdk.org/census#rehn) (@robehn - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/13704/head:pull/13704` \
`$ git checkout pull/13704`

Update a local copy of the PR: \
`$ git checkout pull/13704` \
`$ git pull https://git.openjdk.org/jdk.git pull/13704/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 13704`

View PR using the GUI difftool: \
`$ git pr show -t 13704`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/13704.diff">https://git.openjdk.org/jdk/pull/13704.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/13704#issuecomment-1528009017)

### CODE REVIEW

**Review by coleenp — APPROVED** [review]
Yes, this looks good and also trivial.

File: src/hotspot/share/runtime/threadSMR.cpp [hunk 1]
```diff
@@ -824,10 +824,10 @@ bool ThreadsListHandle::cv_internal_thread_to_JavaThread(jobject jthread,
   // Looks like a live JavaThread at this point.
 
   if (java_thread != JavaThread::current()) {
-    // jthread is not for the current JavaThread so have to verify
-    // the JavaThread * against the ThreadsList.
-    if (EnableThreadSMRExtraValidityChecks && !includes(java_thread)) {
-      // Not on the JavaThreads list so it is not alive.
+    // java_thread is not the current JavaThread so we have to verify it
+    // against the ThreadsList:
```
Comment (dholmes-ora) [inline src/hotspot/share/runtime/threadSMR.cpp:828]: Colon at the end of the comment seems odd

**Review by dholmes-ora — APPROVED** [review]
One minor nit but otherwise good. Thanks.

**Review by sspitsyn — APPROVED** [review]

**Comment by dcubed-ojdk** [comment]
@coleenp, @dholmes-ora and @sspitsyn - thanks for the reviews!

I suppose there isn't any way to give @robehn credit for his reviews when this
fix was part of https://github.com/openjdk/jdk/pull/13519.

File: src/hotspot/share/runtime/threadSMR.cpp (hunk 1 above)
Comment (dcubed-ojdk) [reply src/hotspot/share/runtime/threadSMR.cpp]: Nice catch! Fixed.

**Comment by dcubed-ojdk** [comment]
This fix was tested with Mach5 Tier[1-8] when it was part of:
https://github.com/openjdk/jdk/pull/13519

and is also being tested with Mach5 Tier[1-8] combined with
the fix from JDK-8307068 and what remains in JDK-8305670.

**Review by robehn — APPROVED** [review]
Looks good, thanks!

**Comment by dcubed-ojdk** [comment]
@robehn - Thanks for the re-review!

**Review by dholmes-ora — APPROVED** [review]

**Comment by dcubed-ojdk** [comment]
@dholmes-ora - Thanks for the re-review!

**Pull request closed without merging**